lib/client.dart

```diff
@@ -712,6 +698,18 @@ class Router {
       return new Future.value(true);
     }
 
+    // Check the last route segment to see if its child routes are loaded.
+    // If not, load the child routes, reject the route change in progress,
+    // and re-route with the same path
```

Comment:
i can update the text (I ended up with a slightly different solution after I wrote the comment).  in this case, we're short-circuiting by re-routing after this attempt got past the `preLeave / leave` phase of route processing and before it got to the `preEnter / enter` phase of processing